crates/runtime/README.md

```diff
@@ -0,0 +1,662 @@
+ensures that the function can be called safely within the constraints of the
+runtime.
+
+Finally, the `run()` function invokes the specified Wasm function. During
+execution, any calls to imported functions are handled by the `VMLogic`
+instance, ensuring that all operations adhere to the defined resource limits and
+security policies. If any errors or exceptions occur during execution, they are
+caught, wrapped in appropriate error types, and included in the execution
+outcome.
+
+Throughout the execution process, the `run()` function leverages Wasmer's
+capabilities for sandboxing and resource control. Wasmer provides mechanisms for
+limiting memory usage, controlling CPU usage through "fuel metering", and
+isolating the Wasm execution from the host system. These features, combined with
+the additional controls implemented in the Calimero runtime, ensure a high level
+of security and predictability in the execution of untrusted Wasm code.
+
+After the function execution completes (or if an error occurs), the `run()`
+function collects the results, logs, and any emitted events. It then calls
+`VMLogic::finish()` to package all this information into an `Outcome` struct,
+which provides a comprehensive summary of the execution results.
+
+### VMLogic
+
+The `VMLogic` struct manages the execution context:
+
+  - Holds references to storage and resource limits
+  - Manages registers for temporary data storage
+  - Collects logs and events during execution
+  - Provides methods for interacting with the host environment
+
+The `VMLogic` struct is the centre of the Calimero runtime, orchestrating the
+execution of WebAssembly modules within a controlled and sandboxed environment.
+Its primary purpose is to manage the state, resources, and interactions between
+the guest Wasm code and the host system, ensuring safe and deterministic
+execution.
+
+At its core, `VMLogic` encapsulates several crucial elements: a mutable
+reference to the storage system, an optional [Wasmer][] memory instance, the
+execution context, resource limits, a register system for temporary data
+storage, and collections for logs and events. This comprehensive set of
+components allows `VMLogic` to maintain full control over the execution
+environment, mediating all interactions between the guest code and the host
+system.
+
+One of the key responsibilities of `VMLogic` is to enforce resource limits
+defined in the `VMLimits` struct. These limits cover various aspects of
+execution, including memory usage, stack size, register count and size, and
+constraints on storage operations. By strictly adhering to these limits,
+`VMLogic` prevents potential abuse of resources and ensures that all executions
+remain within predefined boundaries, which is crucial for maintaining the
+integrity and security of the runtime.
+
+The register system implemented within `VMLogic` provides a flexible mechanism
+for temporary data storage and inter-function communication. It allows the guest
```

Comment:
So wait... I don't follow now. I thought in the comment a couple above, you were saying this was not a thing - but - it _is_ a thing? I'm a little confused; I thought I'd got it wrong 😅 

Could you elaborate a little so that I can adjust what I've written?

Reply:
I mean, it's like.. I can either talk to you directly, or I can put it in a note, send it to France, tell you it's in France and you ship it back just to read it.